backend/src/test/java/com/google/sps/MatchDatastoreTest.java

```diff
@@ -0,0 +1,142 @@
+  private static final String PROPERTY_DURATION = "duration";
+  private static final String PROPERTY_TIMESTAMP = "timestamp";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  // TODO(#37): Find/write JUnit rule to encapsulate setUp() and tearDown()
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void addOneMatch() throws EntityNotFoundException {
+    // Add one match to datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    Match match = new Match(PERSON_A, PERSON_B, DURATION_DEFAULT, TIMESTAMP_DEFAULT);
+
+    long id = matchDatastore.addMatch(match);
+
+    Key key = KeyFactory.createKey(KIND_MATCH, id);
+    Entity entity = datastore.get(key);
+    assertThat((String) entity.getProperty(PROPERTY_FIRSTPARTICIPANTUSERNAME)).isEqualTo(PERSON_A);
+    assertThat((String) entity.getProperty(PROPERTY_SECONDPARTICIPANTUSERNAME)).isEqualTo(PERSON_B);
+    assertThat(((Long) entity.getProperty(PROPERTY_DURATION)).intValue())
+        .isEqualTo(DURATION_DEFAULT);
+    assertThat((long) entity.getProperty(PROPERTY_TIMESTAMP)).isEqualTo(TIMESTAMP_DEFAULT);
+  }
+
+  @Test
+  public void addTwoMatches() throws EntityNotFoundException {
+    // Add two matches to datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    Match match1 = new Match(PERSON_A, PERSON_B, DURATION_DEFAULT, TIMESTAMP_DEFAULT);
+    Match match2 = new Match(PERSON_C, PERSON_D, DURATION_DEFAULT, TIMESTAMP_DEFAULT);
+
+    long id1 = matchDatastore.addMatch(match1);
+    long id2 = matchDatastore.addMatch(match2);
+
+    Key key1 = KeyFactory.createKey(KIND_MATCH, id1);
+    Key key2 = KeyFactory.createKey(KIND_MATCH, id2);
+    Entity entity1 = datastore.get(key1);
+    Entity entity2 = datastore.get(key2);
+    assertThat((String) entity1.getProperty(PROPERTY_FIRSTPARTICIPANTUSERNAME)).isEqualTo(PERSON_A);
+    assertThat((String) entity1.getProperty(PROPERTY_SECONDPARTICIPANTUSERNAME))
+        .isEqualTo(PERSON_B);
+    assertThat(((Long) entity1.getProperty(PROPERTY_DURATION)).intValue())
+        .isEqualTo(DURATION_DEFAULT);
+    assertThat((long) entity1.getProperty(PROPERTY_TIMESTAMP)).isEqualTo(TIMESTAMP_DEFAULT);
+    assertThat((String) entity2.getProperty(PROPERTY_FIRSTPARTICIPANTUSERNAME)).isEqualTo(PERSON_C);
+    assertThat((String) entity2.getProperty(PROPERTY_SECONDPARTICIPANTUSERNAME))
+        .isEqualTo(PERSON_D);
+    assertThat(((Long) entity2.getProperty(PROPERTY_DURATION)).intValue())
+        .isEqualTo(DURATION_DEFAULT);
+    assertThat((long) entity2.getProperty(PROPERTY_TIMESTAMP)).isEqualTo(TIMESTAMP_DEFAULT);
```

Comment:
Can you check if the casts are necessary? I suspect they can be removed

Reply:
The casts are necessary. I just checked. getProperty() returns type Object.

Reply:
But can the tests still pass without the casts? Under the hood the assertion will call the .equals() method on the specific concrete Object subclass returned by getProperty, even if the return type is Object.

Reply:
I get this error when I remove the cast: incompatible types: java.lang.Object cannot be converted to java.lang.Str
ing (same for long)